.github/workflows/docs.yml

```
@@ -0,0 +1,26 @@
+name: docs-gen
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Create Documents
+      run: cargo doc --target-dir ./docs/
+
+    - name: Commit files
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git commit -m "Updated Documentation" -a
```

```suggestion
    - name: Commit files
      run: |
        git config --local user.email "action@github.com"
        git config --local user.name "GitHub Action"
        git commit -m "Updated Documentation" -a
```

This is showing that a custom "User" will update our repo with the new documentation if it hasn't been done already.